org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java

```diff
@@ -11,92 +11,177 @@
-            @vector = 1,0,0,
-            @depends_on = "y_centre"
-            @units = "mm"
-            @controller_record = {NX_CHAR} //EPICS name
-        y_centre = {NX_NUMBER}
-            @transformation_type = "translation"
-            @vector = 0,1,0,
-            @depends_on = "."
-            @offset = 0,0,-14500 //This may change
-            @units = "mm"
-            @controller_record = {NX_CHAR} //EPICS name
-    beam:NXbeam
-        //Removed distance from here, since it's in the "y_centre" above
-        incident_beam_divergence[2,i] = {NX_FLOAT}
-            @units = "radians"
-        final_beam_divergence[2,i] = {NX_FLOAT}
-            @units = "radians"
-    motors:NXcollection
-        downstream_x:NXpositioner
-            name = "s1dsX"
-            description = "Downstream X position"
-            value = {NX_NUMBER}
-                @units = "mm"
-            controller_record = {NX_CHAR} //EPICS name
-        downstream_y:NXpositioner
-            name = "s1dsY"
-            description = "Downstream Y position"
-            value = {NX_NUMBER}
-                @units = "mm"
-            controller_record = {NX_CHAR} //EPICS name
-        upstream_x:NXpositioner
-            name = "s1usX"
-            description = "Upstream X position"
-            value = {NX_NUMBER}
-                @units = "mm"
-            controller_record = {NX_CHAR} //EPICS name
-        upstream_y:NXpositioner
-            name = "s1usY"
-            description = "Upstream Y position"
-            value = {NX_NUMBER}
-                @units = "mm"
-            controller_record = {NX_CHAR} //EPICS name
- </pre>
- 
- * 
- * @author Matthew Gerring
- *
+ * @see {@link MockScannableConfiguration}
  */
-public class MockNeXusSlit extends MockNeXusScannable {
+public class MockNeXusSlit extends MockScannable implements INexusDevice<NXslit> {
+
+	public boolean isWritingOn() {
+		return writingOn;
+	}
+
+	public void setWritingOn(boolean writingOn) {
+		this.writingOn = writingOn;
+	}
+
+	private ILazyWriteableDataset xLzSet;
+	private ILazyWriteableDataset yLzSet;
+	private ILazyWriteableDataset xLzValue;
+	private ILazyWriteableDataset yLzValue;
+
+	private boolean writingOn = true;
 
 	public MockNeXusSlit() {
 		super();
 	}
 
+	public MockNeXusSlit(String name, double d, int level) {
+		super(name, d, level);
+	}
 	public MockNeXusSlit(String name, double d, int level, String unit) {
 		super(name, d, level, unit);
 	}
 
-	public MockNeXusSlit(String name, double d, int level) {
-		super(name, d, level);
+	@ScanFinally
+	public void nullify() {
+		xLzSet   = null;
+		xLzValue = null;
+		yLzSet   = null;
+		yLzValue = null;
+	}
+
+	public NexusObjectProvider<NXslit> getNexusProvider(NexusScanInfo info) throws NexusException {
+		final NXslit positioner = NexusNodeFactory.createNXslit();
+
+		if (info.getScanRole(getName()) == ScanRole.METADATA) {
+			//positioner.setField(NXslit.NX_X_GAP, getPosition().doubleValue());

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)